channeldb/migration32/route.go

```diff
@@ -0,0 +1,580 @@
+package migration32
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/wire"
+	lnwire "github.com/lightningnetwork/lnd/channeldb/migration/lnwire21"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// MPPOnionType is the type used in the onion to reference the MPP
+	// fields: total_amt and payment_addr.
+	MPPOnionType tlv.Type = 8
+
+	// AMPOnionType is the type used in the onion to reference the AMP
+	// fields: root_share, set_id, and child_index.
+	AMPOnionType tlv.Type = 14
+)
+
+// VertexSize is the size of the array to store a vertex.
+const VertexSize = 33
+
+// Vertex is a simple alias for the serialization of a compressed Bitcoin
+// public key.
+type Vertex [VertexSize]byte
+
+// Route represents a path through the channel graph which runs over one or
+// more channels in succession. This struct carries all the information
+// required to craft the Sphinx onion packet, and send the payment along the
+// first hop in the path. A route is only selected as valid if all the channels
+// have sufficient capacity to carry the initial payment amount after fees are
+// accounted for.
+type Route struct {
```

Comment:
Oh, I think I just made this a bit more complicated by merging https://github.com/lightningnetwork/lnd/pull/9049, which added a new field to this struct, including new encoding/decoding logic that probably needs to be copied (`lnwire/custom_records.go`). Although no existing records should ever have that data. But if this PR only goes in with 0.19, there could be such records from 0.18.4 to 0.19.

Reply:
This is the relevant commit btw: https://github.com/lightningnetwork/lnd/pull/9049/commits/4804cbf139b38e0f0852ab930575933d56b3e41d